src/server/string_family.cc

```diff
@@ -356,6 +359,92 @@ OpResult<void> SetGeneric(ConnectionContext* cntx, const SetCmd::SetParams& spar
   return cntx->transaction->ScheduleSingleHop(std::move(cb));
 }
 
+OpResult<array<int64_t, 5>> OpThrottle(const OpArgs& op_args, string_view key, int64_t max_burst,
+                                       int64_t count, int64_t period_s, int64_t quantity) {
+  using namespace chrono_literals;
+
+  auto& db_slice = op_args.shard->db_slice();
+
+  const int64_t limit = max_burst + 1;
```

Comment:
I suggest that instead of passing the original arguments into this function, you will pass the derived parameters:
`limit` instead of `max_burst`, `emission_interval_ms` instead of period_s etc. All the preliminary computations 
better do in `ClThrottle` together with the check below
```cpp
if (emission_interval_ms == 0) {
    return OpStatus::OUT_OF_RANGE;
  }
 ```